camel/kie-camel/src/main/java/org/kie/camel/component/KiePolicy.java

```diff
@@ -123,14 +124,36 @@ public static void augmentNodes(RouteContext routeContext, ProcessorDefinition<?
         }
     }
 
-    private static DataFormatDefinition processDataFormatType(RouteContext routeContext, String ref, DataFormatDefinition dformatDefinition) {
-        if (dformatDefinition == null) {
-            if ("json".equals(ref)) {
+    private static DataFormatDefinition processDataFormatType(RouteContext routeContext,
+                                                              String ref,
+                                                              DataFormatDefinition dformatDefinition) {
+        if ( dformatDefinition == null ) {
+
+            // Use Java Reflection to get the method setPermissions. This is done to allow compatibility between fuse 6.3 and fuse 6.2.1
+            // The xstream library differs on version between the 6.2.1 and 6.3 version of fuse.
+            Method setPermissions = null;
+            try {
+                setPermissions = dformatDefinition.getClass().getMethod("setPermissions", String.class);
+            } catch (Exception e) {
+            }
+            if ( "json".equals( ref ) ) {
                 dformatDefinition = new XStreamDataFormat();
-                ((XStreamDataFormat)dformatDefinition).setDriver("json");
-            } else if ("xstream".equals(ref)) {
+                ((XStreamDataFormat) dformatDefinition).setDriver( "json" );
+                if (setPermissions != null) {
+                    try {
+                        setPermissions.invoke(dformatDefinition, "+*");
+                    } catch (Exception e) {
+                    }
```

Comment:
Sorry for late comments, but I think we could extract this into a new method (to avoid duplication), since it is repeated three times in the file?